BootloaderCommonPkg/Library/IppCryptoPerfLib/IppCryptoPerfLib.c

```diff
@@ -0,0 +1,137 @@
+    UINT8 Index;
+    enum Algorith Alg;
+
+    ShellPrint (L"!!! IPP CRYPTO PERFORMANCE MESUREMENT - Pre-Mem!!!  \n");
+    ShellPrint (L" Usecase             | Delta (µs)      \n");
+    ShellPrint (L"---------------------+----------------+\n");
+
+    for (Alg = 0; Alg < MAX_ALGO; Alg++) {
+        for (Index = 0; Index < MAX_ITER; Index++) {
+            if (Alg == SHA256) {
+                    StartTime = ReadTimeStamp();
+                    Sha256 (Data, Iter_premem[Index].Length, DigestBuffer);
+                    StopTime = ReadTimeStamp();
+            } else if (Alg == SHA384) {
+                    StartTime = ReadTimeStamp();
+                    Sha384 (Data, Iter_premem[Index].Length, DigestBuffer);
+                    StopTime = ReadTimeStamp();
+            } else if (Alg == SM3) {
+                    StartTime = ReadTimeStamp();
+                    Sm3 (Data, Iter_premem[Index].Length, DigestBuffer);
+                    StopTime = ReadTimeStamp();
+            }
+            TsTicks  = StopTime - StartTime;
+            ShellPrint (L" %6a for %8a | %7d µs \n",
+                                AlgorithmName(Alg), Iter_premem[Index].Desc,
+                                TimeStampTickToMicroSecond(TsTicks));
+        }
+    }
+    ShellPrint (L"---------------------+----------------+\n");
+}
+
+VOID EFIAPI IppCryptoPerf_Postmem()
+{
+    UINT64 StartTime, StopTime, TsTicks;
+    UINT8 DigestBuffer[MAX_DIGEST_BUFFER_SIZE];
+    UINT8* Data = (UINT8*) AllocatePool (Iter_postmem[MAX_ITER - 1].Length);
+    UINT8 Index;
+    enum Algorith Alg;
+
+    ShellPrint (L"!!! IPP CRYPTO PERFORMANCE MESUREMENT - Post-Mem!!!  \n");
+    ShellPrint (L" Usecase             | Delta (µs)      \n");
+    ShellPrint (L"---------------------+----------------+\n");
+
+    for (Alg = 0; Alg < MAX_ALGO; Alg++) {
+        for (Index = 0; Index < MAX_ITER; Index++) {
+            GenerateRandomNumbers(Data, Iter_postmem[Index].Length);
+            if (Alg == SHA256) {
+                    StartTime = ReadTimeStamp();
+                    Sha256 (Data, Iter_postmem[Index].Length, DigestBuffer);
+                    StopTime = ReadTimeStamp();
+            } else if (Alg == SHA384) {
+                    StartTime = ReadTimeStamp();
+                    Sha384 (Data, Iter_postmem[Index].Length, DigestBuffer);
+                    StopTime = ReadTimeStamp();
+            } else if (Alg == SM3) {
+                    StartTime = ReadTimeStamp();
+                    Sm3 (Data, Iter_postmem[Index].Length, DigestBuffer);
+                    StopTime = ReadTimeStamp();
+            }
+            TsTicks  = StopTime - StartTime;
+            ShellPrint (L" %6a for %8a | %7d µs \n",
+                                AlgorithmName(Alg), Iter_postmem[Index].Desc,
+                                TimeStampTickToMicroSecond(TsTicks));
+        }
+    }
+    ShellPrint (L"---------------------+----------------+\n");
+    FreePool (Data);
+}
```

Comment:
It looks the test functions are same, please check if  we could use a single function.

Reply:
done